src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -339,18 +339,24 @@ private void ChooseActions(in Entity entity, ref MicrobeAI ai, ref CompoundAbsor
             ai.ATPThreshold = 0.95f * speciesFocus / Constants.MAX_SPECIES_FOCUS;
         }
 
-        if (ai.ATPThreshold > 0.0f)
+        if (ai.ATPThreshold > Mathf.Epsilon)
         {
             if (compounds.GetCompoundAmount(atp) < compounds.GetCapacityForCompound(atp) * ai.ATPThreshold)
             {
+                bool outOfSomething = false;
                 foreach (var compound in compounds.Compounds)
                 {
-                    if (IsVitalCompound(compound.Key, compounds) && compound.Value > 0.0f)
+                    if (IsVitalCompound(compound.Key, compounds) && compound.Value <= Mathf.Epsilon)
                     {
-                        control.SetMoveSpeed(0.0f);
-                        return;
+                        outOfSomething = true;
```

Comment:
One idea I have is that you are missing a break here in the loop once the condition is satisfied, which causes a bunch of extra checking once the value is already true and no further iterations of the loop will change it at all.

Reply:
added